app/controllers/episode_media_controller.rb

```diff
@@ -25,6 +25,9 @@ def update
     authorize @episode, :update?
     @episode.assign_attributes(parsed_episode_params)
 
+    # when an uncut is destroyed, also destroy sliced contents
+    @episode.contents.each(&:mark_for_destruction) if @episode.uncut&.marked_for_destruction?
```

Comment:
This is for #942 - opted to fix this in the controller, so it's less magical.  But this fixes the status displayed in the UI, by destroying sliced-contents when the uncut segment they came from is destroyed.